File: api/server/src/main/java/su/plo/voice/api/server/audio/source/ServerEntitySource.java
package su.plo.voice.api.server.audio.source;

import org.jetbrains.annotations.NotNull;
import su.plo.lib.api.server.entity.MinecraftServerEntity;
import su.plo.voice.proto.data.audio.source.EntitySourceInfo;

public interface ServerEntitySource extends ServerPositionalSource<EntitySourceInfo> {

    @NotNull MinecraftServerEntity getEntity();
}

File: api/server/src/main/kotlin/su/plo/voice/api/server/audio/source/ServerEntitySource.kt
package su.plo.voice.api.server.audio.source

import su.plo.slib.api.server.entity.McServerEntity
import su.plo.voice.proto.data.audio.source.EntitySourceInfo

/**
 * Represents an entity audio source.
 */
interface ServerEntitySource : ServerProximitySource<EntitySourceInfo> {

    /**
     * Gets the entity associated with this audio source.
     *
     * @return The entity.
     */
    val entity: McServerEntity
}
